### PR TITLE
[WFLY-10694] Fix of WFCORE-3826 breaks plain authentication for ejbs …

### PR DESCRIPTION
**Pull request by JiriOndrusek**
…using legacy configuration - test

Issue: https://issues.jboss.org/browse/WFLY-10694

Test for core issue: https://issues.jboss.org/browse/WFCORE-3963

Needs changes from core: https://github.com/wildfly/wildfly-core/pull/3391

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

**Comment by JiriOndrusek** [comment]
Do not merge it yet, test will be  upgraded according to the last change from https://github.com/wildfly/wildfly-core/pull/3391

**Comment by kabir** [comment]
Adding the hold label as per @JiriOndrusek 's comments

**Comment by JiriOndrusek** [comment]
@kabir PR is good to go, I've managed to extend test on Friday afternoon. Hold label could be removed.

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
@JiriOndrusek The elytron job fails in the added test

**Comment by JiriOndrusek** [comment]
@kabir I'm fixing failing tests. Please do not merge it without my confirmation (I'm not able to simulate error on my local machine, so I'll try a few test runs before merge)

**Comment by JiriOndrusek** [comment]
Retest this please

**Comment by JiriOndrusek** [comment]
Retest this please

**Comment by JiriOndrusek** [comment]
Retest this please

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/SaslLegacyMechanismConfigurationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+         }
+
+         ModelNode response = execute(compositeOperation, mcc);
+         Assert.assertEquals(response.toString(), SUCCESS, response.get(OUTCOME).asString());
+
+         ServerReload.reloadIfRequired(managementClient.getControllerClient());
+      }
+
+      @Override
+      public void tearDown(ManagementClient managementClient, String containerId) throws Exception {
```
Comment (kabir) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/security/SaslLegacyMechanismConfigurationTestCase.java]: @JiriOndrusek I think you can get rid of this method if you extend SnapshotRestoreSetupTask instead

**Comment by JiriOndrusek** [comment]
@kabir thanks, I'll try it. I've noticed this setupTask last week (before it was merged) but I've forgotten to use it in this PR.

**Comment by JiriOndrusek** [comment]
@kabir It's done

**Pull request merged**